src/System.IO.FileSystem/src/System/IO/Win32FileSystemEnumerable.cs

```diff
@@ -454,12 +456,32 @@ private void AddSearchableDirsToStack(SearchData localSearchData)
             }
         }
 
-        private static String NormalizeSearchPattern(String searchPattern)
+        // This is faster than using List<T>.Reverse because it avoids
+        // boxing in List<T>.Reverse's implementation due to the use of
+        // a custom struct for T.
+        private void ReverseSearchList(int index)
```

Comment:
I was going to just go ahead and update this PR to remove it now under the assumption that dotnet/coreclr#1231 would be merged. Unless you think it'd be better to wait a little bit for the coreclr change to propagate?